pkg/operator/mon/health.go

```diff
@@ -75,9 +75,44 @@ func (c *Cluster) checkHealth() error {
 	}
 	logger.Debugf("Mon status: %+v", status)
 
-	// failover the unhealthy mons
+	// Source of thruth of which mons should exist is our *clusterInfo*
+	monsNotFound := map[string]interface{}{}
+	for _, mon := range c.clusterInfo.Monitors {
+		monsNotFound[mon.Name] = struct{}{}
+	}
+
+	// variable to track if a failover is happening
+	failoverHappening := false
+
+	// first handle mons that are not in quorum but in the cecph mon map
+	//failover the unhealthy mons
 	for _, mon := range status.MonMap.Mons {
 		inQuorum := monInQuorum(mon, status.Quorum)
+		// if the mon is in quorum remove it from our check for "existence"
+		//else see below condition
+		if _, ok := monsNotFound[mon.Name]; ok {
+			delete(monsNotFound, mon.Name)
+		} else {
+			// when the mon isn't in the clusterInfo, but is in qorum and there are
+			//enough mons, remove it else remove it on the next run
+			if inQuorum && len(status.MonMap.Mons) > c.Size {
+				c.removeMon(mon.Name)
+			} else {
+				logger.Warningf(
+					"mon %s is TOO MUCH in quorum, not enough mons to remove now (wanted: %d, current: %d)",
```

Comment:
Is this an unhealthy mon? what does it mean that the mon is `too much in quorum`?

Reply:
@travisn `too much in quorum` means that there is for some reasons a mon in the Ceph mon map that isn't in the source of truth (anymore and didn't got properly removed).
This removes "rogue" mons that are too much.

Normal `true` condition [health.go L99](https://github.com/rook/rook/pull/1080/files/cc5d7200abdb07e37ea29593448c3c8fd5b1fc39#diff-bde5ec138117f3d7d9ee4274646be734R99) removes the mon that is too much.
The `else` condition means that there aren't enough "known" mons in the Ceph mon map (see [health.go L98](https://github.com/rook/rook/pull/1080/files/cc5d7200abdb07e37ea29593448c3c8fd5b1fc39#diff-bde5ec138117f3d7d9ee4274646be734R98)) for it to be removed.

I changed the comments to better reflect that now.